src/libraries/Pausable.sol

```diff
@@ -71,19 +118,27 @@ abstract contract Pausable is Initializable {
         _;
     }
 
-    function __Paused_init() internal onlyInitializing {
-        __Paused_init_unchained();
+    /*
+     * @dev Modifier to allow only the Pauser to access some functionality
+     */
+    modifier onlyPauser() {
+        _checkPauser();
+        _;
     }
 
-    function __Paused_init_unchained() internal onlyInitializing {
-        PauseStorage storage $ = _getPauseStorage();
-        $._pauseFlag = NOT_PAUSED;
+    /*
+     * @dev Modifier to allow only the Pauser to access some functionality
+     */
+    function _checkPauser() internal view {
+        if (_pauser() != msg.sender) {
+            revert InvalidPauser(msg.sender);
+        }
     }
 
     /**
      * @dev pauses the function and emits the `Paused` event
      */
-    function _pause() internal virtual whenNotPaused {
+    function _pause() internal virtual whenNotPaused onlyPauser {
```

Comment:
I thought we wanted both the owner and pauser to be able to pause? You'll need to create a helper modifier and probably use that with the `external` pause methods in the inheriting contracts

Reply:
my thinking for now was just to collapse the pauser role to owner for now given we don't expect the pauser to be distinct from the owner. (although it is a simple change - we could extend it now).

Reply:
I would personally vote for extending that now as it's not a heavy lift but we can discuss in the standup later if needed

Reply:
sure

Reply:
I'm also for having the owner and pauser be able to pause.
We could have an `onlyPauser` modifier and another `onlyPauserAndOwner(address owner)` modifier?